src/opera_utils/disp/search.py

```diff
@@ -152,6 +152,7 @@ def get_products(
     product_version: str | None = None,
     start_datetime: datetime | None = None,
     end_datetime: datetime | None = None,
+    url_type: Literal["s3", "https"] = "https",
```

Comment:
In general, I think you don't want to use `Literal` for input parameters like this.

`Literal` only accepts literals -- not variables -- so something like this will cause mypy to complain:

```python
url_type = "https"
get_products(..., url_type=url_type, ...)
```

I think `Literal` is more useful for return values. For example, if you had a function (or function overload) that always returns `True`, you could annotate the return type as `Literal[True]` instead of `bool`.

A better option here is probably to use a [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) (or just an `Enum` that also inherits from `str` in order to support Python<3.11)